lib/main.js

```diff
@@ -44,6 +44,10 @@ module.exports = {
     return this.autocompleteManager
   },
 
+  provideAutocompleteManagerCaller () {
```

Comment:
See how the `consumeFoo` services are provided below? It seems like something like `provideManager` could blend in well with those. I see where you're coming from wanting the name to document that it's a promise. I feel like that's an issue in a lot of places when returning promises, but I think it really only makes sense to distinguish when there are both sync and async variants of a method. In that case, I think the norm in JS is to decorate the synchronous version with `sync`.

So, to be clear, when the service is consumed it will actually be a promise? It's unfortunate that we didn't design the system to automatically wait on promises before calling consumers in situations like this.